openedx/event-bus-kafka
# feat: remove confluent-kafka as a hard dependency

**Description:** 
Remove confluent-kafka as a dependency in base.in. Instead, wrap all imports in a try...catch and log error messages if the library is not found.

**Issue:** 
https://github.com/openedx/openedx-events/issues/88

**Installation instructions:**
As of this PR, any service that uses event-bus-kafka will need to independently install confluent-kafka. 

**Testing instructions:**

Follow the instructions in manual_testing.rst EXCEPT for the bit about installing confluent-kafka.
You should get `Library confluent-kafka not available. Cannot create event producer.` (or consumer).

In the cms shell, `pip install confluent-kafka`
You should then be able to go through manual_testing again and get the correct results.

`make test` in the event-bus-kafka repo itself should always work (without having to install confluent-kafka separately) since confluent-kafka is in the test dependencies.

**Merge checklist:**

**Post merge:**
      finished.

**Author concerns:** 
Deviated a bit from @robrap 's example of newrelic, where everything is in an `if newrelic` block. Instead, I used more early returns since I find that easier to read (rather than indenting entire methods). This does have the downside of making coverage.py upset since I have not been able to unit test those blocks. I'm not yet certain whether it's possible or whether we should just `pragma: no-cover` it or whether we should ignore it because the coverage check isn't mandatory.
